mcache/tests/common.py

```diff
@@ -1,10 +1,22 @@
 # (C) Datadog, Inc. 2010-2017
 # All rights reserved
 # Licensed under Simplified BSD License (see LICENSE)
+
+import os
+
 from datadog_checks.utils.common import get_docker_hostname
 
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+ROOT = os.path.dirname(os.path.dirname(HERE))
+
 PORT = 11211
 SERVICE_CHECK = 'memcache.can_connect'
 HOST = get_docker_hostname()
 USERNAME = 'testuser'
 PASSWORD = 'testpass'
+
+SOCKET = '/tmp/memcached.sock'
+LOCAL_TMP_DIR = os.path.join(ROOT, "tmp")
```

Comment:
Let's rework the logic to use `tempdir`. Example: https://github.com/DataDog/integrations-core/blob/master/directory/tests/test_bench.py

Reply:
Yeah i tried that but docker doesn't seem to mount folders created this way. I am getting error like this one `ERROR: for compose_memcached_socket_1  Cannot start service memcached_socket: b'Mounts denied: \r\nThe path /var/folders/c_/4msvxd_n67g3dcm5xz42bcgh0000gn/T/tmpWhC8xL/mcache/memcached.sock\r\nis not shared from OS X and is not known to Docker.\r\nYou can configure shared paths from Docker -> Preferences... -> File Sharing.\r\nSee https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.\r\n.'`

Reply:
use `os.path.realpath()` and it should work